inflight_batcher_llm/src/libtensorrtllm.cc

```diff
@@ -609,8 +609,17 @@ class WorkItemsQueue
         // Check if work item has been stopped
         if (mStoppedReqIds.find(workItem->requestId()) == mStoppedReqIds.end())
         {
-            mInProgressWorkItems.emplace(std::make_pair(workItem->requestId(), workItem));
-            markedInProgress = true;
+            // Check if the Triton request behind is cancelled
```

Comment:
would it make sense to move this to before the if statement?
```
            // Check if the Triton request behind is cancelled
            bool is_cancelled = false;
            TRITONBACKEND_ResponseFactoryIsCancelled(workItem->response_factory(), &is_cancelled);
            if ((!is_cancelled) && (mStoppedReqIds.find(workItem->requestId()) == mStoppedReqIds.end()))
```

Reply:
Good idea! Updated